daemon/cluster/provider/cluster.go

```diff
@@ -0,0 +1,36 @@
+package provider
```

Comment:
This isn't really the correct use of the word "provider". Typically, a provider emits instances of a thing. For example, they would be used in a dependency injection scenario to "provide" a cluster.

In Go, I would expect provider interface to be like this:

```
type ClusterProvider interface {
  Get() Cluster
}
```

Now, I don't think this the intent of this PR or this particular package.